### PR TITLE
Added doc for Gitlab pipelines integration

### PR DESCRIPTION
**Pull request by e-stadnik**
It's documentation part of pull request https://github.com/gitextensions/gitextensions/pull/11105 for Gitlab pipelines integration

### CODE REVIEW

File: source/settings.rst [hunk 1]
```diff
@@ -559,6 +559,20 @@ This page contains general settings for Git Extensions.
...
+  
+	Enter the URL of the server (and port, if applicable).
+	
+  .. setting:: Project ID
+  
+	Use ``Get Project ID from server`` link to obtaint it automatically (required valid access token or public access rights for project). Alternativelly you can enter the Project ID from Gitlab instance website.
+	
+  .. setting:: Api Token
+  
+	Open ``Edit Profile`` -> ``Access Tokens`` menu on Gitlab instance website to generate new access token. Required ``read_api`` permission. Can be empty for projects with public access rights.
```
Comment (RussKie) [inline source/settings.rst:575]: Could you please use spaces instead of tabs?

**Review by RussKie — COMMENTED** [review]
👍

File: source/settings.rst [hunk 2]
```diff
@@ -559,6 +559,20 @@ This page contains general settings for Git Extensions.
...
+    
+.. settingsgroup:: Gitlab
+
+  .. setting:: Gitlab server URL
+  
+    Enter the URL of the server (and port, if applicable).
+    
+  .. setting:: Project ID
+  
+    Use ``Get Project ID from server`` link to obtaint it automatically (required valid access token or public access rights for project). Alternativelly you can enter the Project ID from Gitlab instance website.
```
Comment (gerhardol) [inline source/settings.rst:571]:
```suggestion
    Use ``Get Project ID from server`` link to obtain it automatically (required valid access token or public access rights for project). Alternatively you can enter the Project ID from Gitlab website.
```

**Comment by e-stadnik** [comment]
@gerhardol please merge it to main, because it's missing in 4.2 documentations 
https://git-extensions-documentation.readthedocs.io/en/release-4.2/settings.html#build-server-integration

**Comment by gerhardol** [comment]
Thanks!
Sorry that I missed this. The doc need some more love, as usual (but there were some updates for 4.1).
Doc rebuilt for main and 4.2

**Pull request merged**